launch/launcher.go

```diff
@@ -124,6 +146,85 @@ func (l *Launcher) env(process Process) error {
 	})
 }
 
+func (l *Launcher) execD(process Process) error {
```

Comment:
What do you think about changing the function name to `launchWithExecutable` in order to be consistent with `launchWithShell` (from `shell.go`)?

Reply:
This is out of date after 4147c88 but I think it is meaningfully different from `launchWithShell` because `launchWithShell` actually launches the process whereas `execD` (now `doExecD` only executes the `exec.d` binaries).